datapath: Remove 2005 route table for IPv4

jschwinger233 opened the pull request:
This reverts 3ed62d51d5ac27bbe2c60372850d4ec78c17bb6e for IPv4 part only, as issue #21954 has been resolved by #24208.

Another PR #24882 removes 2005 route table for IPv6.

Signed-off-by: Zhichuan Liang <gray.isovalent.com>

```release-note
Fix broken IPv4 connectivity from outside to NodePort service when using L7 ingress policy, by removing PROXY_RT route table.
```

jschwinger233 committed:
datapath: remove 2005 route table for ipv4 only

This reverts commit 3ed62d5 partially and only removes ipv4 2005 route
table.

Fixes: #21954

Signed-off-by: Zhichuan Liang <gray.liang@isovalent.com>

pchaigno reviewed (approved):
LGTM.

> This PR removes 2005 route table for IPv6, and changes datapath for return IPv6 traffic from L7 proxy.

Same comment as for the IPv6 PR:
Our release notes usually don't have a subject; it is understood to be "this release". Also nit: the comma before "and" is superfluous since both clauses share the same subject.

ti-mo commented on bpf/init.sh:
```diff
@@ -108,8 +108,7 @@ function move_local_rules()
 {
-	# Any packet from an ingress proxy uses a separate routing table that routes
-	# the packet back to the cilium host device.
+	# TODO(brb): remove $PROXY_RT_TABLE -related code in v1.14
```
Is this comment still relevant? This is going into 1.14, so maybe this should be changed to 1.15?

jschwinger233 committed:
test: add e2e test for reaching nodeport from outside with L7 policy

This commit adds e2e test to cover issue #21954.

Test cases for IPv6 are deleted and PR #24882 will take care of them.

Signed-off-by: Zhichuan Liang <gray.liang@isovalent.com>

julianwiedmann commented on test/k8s/services.go:
```diff
@@ -289,7 +289,7 @@ var _ = SkipDescribeIf(helpers.RunsOn54Kernel, "K8sDatapathServicesTest", func()
 			It("Tests NodePort with L7 Policy", func() {
 				applyPolicy(kubectl, demoPolicyL7)
-				testNodePort(kubectl, ni, false, false, 0)
+				testNodePort(kubectl, ni, false, true, 0)
```
shouldn't such "from-outside" test be in the `Checks N/S loadbalancing` section below? :thinking:

christarazi commented on test/k8s/services.go:
```diff
@@ -644,6 +644,28 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`,
 			})
 		})
 
+		SkipContextIf(func() bool {
+			return helpers.DoesNotRunWithKubeProxyReplacement() || helpers.RunsOnAKS() || helpers.DoesNotExistNodeWithoutCilium()
+		}, "with L7 policy", func() {
+			var demoPolicyL7 string
+
+			BeforeAll(func() {
+				demoPolicyL7 = helpers.ManifestGet(kubectl.BasePath(), "l7-policy-demo.yaml")
+			})
+
+			AfterAll(func() {
+				kubectl.Delete(demoPolicyL7)
+				// Same reason as in other L7 test above
+				kubectl.CiliumExecMustSucceedOnAll(context.TODO(),
+					"cilium bpf ct flush global", "Unable to flush CT maps")
+			})
+
+			It("Tests NodePort with L7 Policy from outside", func() {
+				applyPolicy(kubectl, demoPolicyL7)
+				testNodePort(kubectl, ni, false, true, 0)
+			})
+		})
```
Is this test covered by what's in the CLI connectivity tests or the ci-datapath job?

jschwinger233 replied:
Just checked cilium-cli, there is connectivity test for reaching nodeport from outside, but no L7 policy is applied there to cover #21954: 
https://github.com/cilium/cilium-cli/blob/91329434f04b605e19eb329b8e891f0aa1df49a0/connectivity/tests/service.go#L208-L217
I understand we plan to abandon e2e test suites, I'll open another PR for cilium-cli to cover this scenario after this gets merged.
